src/login/logind-dbus.c

```diff
@@ -2309,6 +2309,24 @@ static int setup_wall_message_timer(Manager *m, sd_bus_message* message) {
         return 0;
 }
 
+static int cache_wakeup_count(void) {
+        static const char kernel_wakeup_count[] = "/sys/power/wakeup_count";
+        static const char systemd_wakeup_count[] = "/run/wakeup_count";
+        _cleanup_free_ char *buf = NULL;
+        size_t bufsize;
+        int r;
+
+        r = read_full_virtual_file(kernel_wakeup_count, &buf, &bufsize);
```

Comment:
specify the last argument as NULL if you are not interested in it.

Note that if you pass NULL as last arg to read_full_virtual_file() the function will make an extra check: ensure that the returned buffer does not contain a NUL byte. Thus, by passing the third argument as NUL an extra safety check is employed that ensures you can actually use the thing as C string safely. This matters, because below you treat it as a regular NUL terminated C string, when you call write_string_file()